lib/datadog/tracing/contrib/rack/middlewares.rb

```diff
@@ -54,16 +54,18 @@ def call(env)
             # Find out if this is rack within rack
             previous_request_span = env[Ext::RACK_ENV_REQUEST_SPAN]
 
+            return @app.call(env) if previous_request_span
```

Comment:
Does this work okay with AppSec? We're now short circuiting Rack instrumentation whenever there was a Rack span before. My understanding was AppSec produces its own Rack span as well... would it get skipped? CC @lloeki

Reply:
Oh my I think this is going to be a problem. In fact I think the whole PR might be.

Reply:
After discussion, it should be fine with a passing test suite.